sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/TableEntity.java

```diff
@@ -0,0 +1,50 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+package com.azure.data.tables;
+
+import java.util.Map;
+
+/**
+ * table entity class
+ */
+public class TableEntity {
+    private Map<String, Object> properties;
+    //tableName
+    //etag
+
+    /**
+     * creates a new TableEntity
+     *
+     * @param properties map of properties of the entity
+     */
+    public TableEntity(Map<String, Object> properties) {
+    }
+
+    /**
+     * returns a map of properties
+     *
+     * @return map of properties of thsi entity
+     */
+    public Map<String, Object> getProperties() {
+        return properties;
+    }
+
+    /**
+     * adds a new property to this entity's property map
+     *
+     * @param key the key of the property
+     * @param value the value of the property
+     */
+    public void addProperty(String key, Object value) {
+
+    }
+
+    /**
+     * set the properties
+     *
+     * @param properties properties to set to this entity
+     */
+    public void setProperties(Map<String, Object> properties) {
```

Comment:
This and the one above are similar. Do we expect users to pass an entire map, or chain a bunch of addProperty calls?

I'd prefer addProperty (since it is used in your samples) and remove this one. Also, addProperties should be fluent, and return `this` like your query options class. Also, annotate class with `@Fluent`